### ISSUE TITLE
[bug] '/' path separator is not handled correctly on Windows by Conan 1.30.0

### ISSUE THREAD

**Issue by db4** [conan-io/conan#7856]
### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10
  * Conan version: 1.30.0
  * Python version: 3.8.6

### Steps to reproduce
The following consumer recipe illustrates the problem:
```python
from conans import ConanFile
import os

class TestConan(ConanFile):
    requires = "zlib/1.2.11@conan/stable"

    def imports(self):
        pkg = "zlib"
        path = "licenses/LICENSE"
        self.copy(path, dst=pkg, root_package=pkg)
        if not os.path.exists(os.path.join(pkg, path)):
            raise Exception("Import failure!")
```
It works with `1.29.2` but fails with `1.30.0`. If I replace `/` with `\\` it works with `1.30.0` as well.

**Comment by DoDoENT** [comment]
I was just about to report that. In my case, the following does not work in my conan package:

```python
    def imports(self):
        self.copy('some-folder/*', root_package='some-other-package')
```

The `some-folder` gets imported on mac and Linux, but not on Windows. Downgrade to conan v1.29.2 solved the issue.

**Comment by czoido** [comment]
Hi @db4, @DoDoENT,
Thanks a lot for reporting. I have been able to reproduce the issue, it was probably introduced by https://github.com/conan-io/conan/pull/7704
We'll fix as soon as possible.

**Comment by memsharded** [comment]
Fixed by https://github.com/conan-io/conan/pull/7861, will be released in 1.30.2
